docs/www/data-transform.md

```diff
@@ -0,0 +1,393 @@
+---
+title: Data transformation
+order: 0
+---
+# Data transformation
+
+The basic process of streaming data is to have systems produce events to a topic
+and have other systems consume from that topic.
+What if you need to change the content of the events, or transform the events, 
+in some way like scrubbing private data?
+Normally you'd have to set up another process to consume the events, transform them,
+and produce them to another topic that your consumers can consume from.
+That's extra overhead of code, integration, monitoring, and performance load that you'd like to avoid if possible.
+
+With Redpanda, you can now deploy custom Node.js programs to handle the entire consume-transform-produce cycle for you.
+
+We do all of this co-processing using a new transformation engine built on-top of [WASM (WebAssembly)](https://developer.mozilla.org/en-US/docs/WebAssembly) technology.
+
+All you need to do is to provide:
+
+- Your Javascript program that transforms the data. 
+- Specify what source topics the program will receive data from.
+- The destination topics where produced data will be stored. 
+
+You can jump start your data transformation development with `rpk`. We provide a sample Node.js project so you can start developing right away. Then you can deploy the data transform to Redpanda for all of the brokers to run.
+
+Be aware that even though we're referring to this feature as data transformation, the Kafka topics are still immutable. 
+What happens under the hood, is that we're reading from an immutable Kafka topic, applying a data transformation process and then we producing to another immutable Kafka topic.
+Your data is never actually transformed, but rather read and written again. 
+
+**THIS FEATURE IS A TECHNICAL PREVIEW** - We don't provide official support for it nor do we recommend its use in production. 
+
+To get up-to-date with what we're doing right now, and to meet other Redpanda users you can join [our Slack community](https://join.slack.com/t/vectorizedcommunity/shared_invite/zt-ng2ze1uv-l5VMWSGQHB9gp47~kNnYGA).
```

Comment:
Do we commonly call out a link to our Slack community in docs feature pages? If not, can remove.